DB/src/main/java/io/deephaven/stream/StreamToTableAdapter.java

```diff
@@ -29,11 +54,83 @@ public StreamToTableAdapter(@NotNull final TableDefinition tableDefinition,
         this.liveTableRegistrar = liveTableRegistrar;
         streamPublisher.register(this);
         liveTableRegistrar.addTable(this);
-        // TODO: Construct table using the supplied definition and a series of chunk-wrapping `SwitchColumnSource`s.
-        table = null;
+
+        columnTypesArray = tableDefinition.getColumnTypesArray();
+        columnNamesArray = tableDefinition.getColumnNamesArray();
+
+        chunkTypes = tableDefinitionToChunkTypes(columnTypesArray);
+        currentChunkSources = makeChunkSources(columnTypesArray, columnNamesArray);
+        nullColumnSources = makeNullColumnSources(currentChunkSources);
+
+        final LinkedHashMap<String, ColumnSource<?>> visibleSources = new LinkedHashMap<>();
+        switchSources = makeSwitchSources(currentChunkSources, visibleSources);
+
+        bufferChunkSources = makeChunkSources(columnTypesArray, columnNamesArray);
+
+        index = Index.FACTORY.getEmptyIndex();
+
+        table = new QueryTable(index, visibleSources);
+        table.setRefreshing(true);
+        table.setAttribute(Table.STREAM_TABLE_ATTRIBUTE, Boolean.TRUE);
+    }
+
+    @NotNull
+    private ChunkColumnSource<?> [] makeChunkSources(Class<?>[] columnTypesArray, String[] columnNamesArray) {
+        final ChunkColumnSource<?> [] sources = new ChunkColumnSource[columnNamesArray.length];
+        final TLongArrayList offsets = new TLongArrayList();
+        for (int ii = 0; ii < columnTypesArray.length; ++ii) {
+            sources[ii]  = ChunkColumnSource.make(chunkTypes[ii], columnTypesArray[ii], offsets);
+        }
+        return sources;
+    }
+
+    @NotNull
+    private NullValueColumnSource<?> [] makeNullColumnSources(ChunkColumnSource<?> [] chunkSources) {
+        final NullValueColumnSource<?> [] sources = new NullValueColumnSource[chunkSources.length];
+        final TLongArrayList offsets = new TLongArrayList();
+        for (int ii = 0; ii < columnTypesArray.length; ++ii) {
+            sources[ii]  = NullValueColumnSource.getInstance(chunkSources[ii].getType(), chunkSources[ii].getComponentType());
+        }
+        return sources;
     }
 
-    public Table table() {
+    @NotNull
+    private SwitchColumnSource [] makeSwitchSources(ChunkColumnSource<?> [] wrapped, Map<String, ColumnSource<?>> visibleSourcesMap) {
+        final SwitchColumnSource [] switchSources = new SwitchColumnSource[wrapped.length];
+        final Map<String, SwitchColumnSource<?>> result = new LinkedHashMap<>();
+        for (int ii = 0; ii < wrapped.length; ++ii) {
+            final SwitchColumnSource<?> switchSource = new SwitchColumnSource<>(wrapped[ii]);
```

Comment:
Change the constructor for `SwitchColumnSource` to use the super constructor that takes component type, please. This method can be static, like the others.